data/blocks.yml

```diff
@@ -0,0 +1,36 @@
+uart:
+  instances: 5
+  ios:
+    - name:    tx
+      type:    output
+    - name:    rx
+      type:    input
+      default: 1
+i2c:
+  instances: 2
+  ios:
+    - name:    scl
+      type:    in_out
+      combine: and
+    - name:    sda
+      type:    in_out
+      combine: and
+spi:
+  instances: 7
```

Comment:
With the pinmux we should seek to reduce the number of SPI and UART instances, we've got so many as we've needed one per physical interface.

Need to think carefully about i2c as well, it's a big block so good resource savings by reducing the number. Though still if we're not using the resources for anything else may as well have more i2c, 2 feels like a reasonable number though we could likely manage with 1.

Reply:
I've captured this here: https://github.com/lowRISC/sonata-system/issues/169